components/core/src/clp/NetworkReader.hpp

```diff
@@ -103,7 +105,9 @@ class NetworkReader : public ReaderInterface {
             std::chrono::seconds connection_timeout
             = CurlDownloadHandler::cDefaultConnectionTimeout,
             size_t buffer_pool_size = cDefaultBufferPoolSize,
-            size_t buffer_size = cDefaultBufferSize
+            size_t buffer_size = cDefaultBufferSize,
+            std::map<std::string, std::string> const& custom_headers
+            = std::map<std::string, std::string>()
```

Comment:
1. We should use `std::unordered_map` instead of `std::map`. In C++, unordered map is a hash-map-based data structure where the map is a binary search tree. We use a map only if the order matters.
2. We should pass an optional unordered map instead, and set the default value to `std::nullopt`. The overall signature should be like the following:
```C++
std::optional<std::unordered_map<std::string, std::string>> custom_headers
```
notice that we don't use const reference here because we prefer [modernize-pass-by-value](https://releases.llvm.org/16.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/modernize/pass-by-value.html)

Reply:
fixed in next commit

Reply:
fixed in next comment